src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs

```diff
@@ -148,7 +148,10 @@ internal bool CanGenerateLocal()
             internal bool CanGenerateParameter()
             {
                 // !this.IsInMemberContext prevents us offering this fix for `x.goo` where `goo` does not exist
-                return ContainingMethod != null && !IsInMemberContext && !IsConstant;
+                return ContainingMethod is { IsImplicitlyDeclared: false }
+                    // Workaround: The compiler returns IsImplicitlyDeclared = false for <Main>$.
+                    && ContainingMethod.Name != WellKnownMemberNames.TopLevelStatementsEntryPointMethodName
+                    && !IsInMemberContext && !IsConstant;
```

Comment:
extrac helper for common code here and above.

Reply:
@CyrusNajmabadi The checks are different. One is checking for `TopLevelStatementsEntryPointMethodName` and the other is checking for `TopLevelStatementsEntryPointTypeName` the only common check is for IsImplicitlyDeclared.

Reply:
thanks!